documentation/docs/examples/point-to-point-measurement.mdx

```diff
@@ -0,0 +1,112 @@
+---
+id: point-to-point-measurement
+title: Point to Point Measurement
+description: Demo shows how to measure point to point distance in Reveal.
+---
+
+import { DemoWrapper } from '@site/docs/components/DemoWrapper';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+The `MeasurementTool` is a measuring tool for point to point distance with the 3D objects.
+This can be achieved by selecting a point on the 3D Object and drag the pointer to the destined point to get measurement of the distance.
+The tools default measurement is in meters, but the tool provides user to customise the measuring units.
+
+The tool provides ability to change measuring line width & color.
+
+<DemoWrapper name="Cognite3DViewerDemo" />
+
+## Point to point distance measurment
+To enter into point to point measurement mode, use `MeasurementTool.enterMeasurementMode()` which will have default line width & color.
+
+```jsx runnable
+// import { MeasurementTool } from '@cognite/reveal/tools';
+
+const measurementTool = new MeasurementTool(viewer);
+measurementTool.enterMeasurementMode();
+```
+### Exit measurement
+To exit from measurement mode, use `MeasurementTool.exitMeasurementMode()`.
+
+```jsx runnable
+// import { MeasurementTool } from '@cognite/reveal/tools';
+
+const measurementTool = new MeasurementTool(viewer);
+measurementTool.enterMeasurementMode();
+measurementTool.exitMeasurementMode();
+```
+
+### Set width and color of the measuring line
+To change width and color of the line, use `MeasurementTool.setLineOptions(options: MeasurementOptions, meshObject?: THREE.Mesh)`. MeasurementOptions expects
+  - `lineWidth` : number
+  - `color` : number
+
+```jsx runnable
+// import { MeasurementTool } from '@cognite/reveal/tools';
+
+const lineOptions = {
+  lineWidth: 3,
+  color: 0xff0000
+};
+const measurementTool = new MeasurementTool(viewer);
+measurementTool.enterMeasurementMode();
+measurementTool.setLineOptions(lineOptions);
+```
+
+It's also possible to apply custom label content for the measurment.
+The tool provides option of custom callback for the users. `MeasurementOptions.distanceToLabelCallback` can be
+used to trigger custom operation like changing the measuring units to feet:
+
+```jsx runnable
+// import { MeasurementTool } from '@cognite/reveal/tools';
+
+const measurementTool = new MeasurementTool(viewer, {distanceToLabelCallback: (distanceInMeters) => {
+      // 1 meter = 3.281 feet
+      return `${(distanceInMeters * 3.281).toFixed(2)} ft`;
+     }});
+measurementTool.enterMeasurementMode();
+```
+
+### Get all measurement
+To retrive all measurements added in the viewer, we can provide `MeasurementTool.getAllMeasurement()` which will
+returns array of meshes `THREE.Mesh[]`.
+
+```jsx
+// import { MeasurementTool } from '@cognite/reveal/tools';
+
+const measurementTool = new MeasurementTool(viewer);
+measurementTool.enterMeasurementMode();
+
+//get all measurements
+const measurements = measurementTool.getAllMeasurement();
```

Comment:
Do "useful" stuff, this example doesn't do anything and there will not be any measurements available and the result is just discarded. E.g., when user has done a measurement - show an `alert` with the distance.